mongorestore/metadata.go

```diff
@@ -229,6 +229,17 @@ func (restore *MongoRestore) LegacyInsertIndex(intent *intents.Intent, index Ind
 	return nil
 }
 
+// UpdateAutoIndexId updates {autoIndexId: false} to {autoIndexId: true} if the server version is
+// >= 4.0 and the database is not `local`.
+func (restore *MongoRestore) UpdateAutoIndexId(options bson.D) {
```

Comment:
Based on the docstring for this function, it looks like we should actually have the version check inside here instead of at the call sites.

Alternatively, if you'd prefer, you could keep the version check outside of this function; in that case, however, I don't think there would be any need for this to be a method on `MongoRestore` (it could just be a plain helper function)

Reply:
True! Doing so also eliminates some duplicate code.